sklearn/impute/__init__.py

```diff
@@ -1,8 +1,16 @@
 """Transformers for missing value imputation"""
+import typing
 
 from ._base import MissingIndicator, SimpleImputer
 from ._knn import KNNImputer
 
+if typing.TYPE_CHECKING:
+    # Workaround for type checkers (e.g. mypy) to avoid
+    # import errors for experimenal estimators.
+    # TODO: remove the above check once the estimator is no longer
+    #       experimental.
+    from ._iterative import IterativeImputer  # noqa
+
```

Comment:
Should we add this this to where we explain how to have an experimental module? is it in `maintainer.rst`?

Reply:
The section on experimental estimators, says to copy and adapt the code from HistGradientBoosting and/or Iterative regressor, which should be no need to more specific documentation, I think.